tests/test_other.py

```diff
@@ -9875,3 +9875,9 @@ def test_emranlib(self):
     run_process([PYTHON, EMRANLIB, 'liba.a'])
     output = run_process([shared.LLVM_NM, '--print-armap', 'liba.a'], stdout=PIPE).stdout
     self.assertContained('Archive map', output)
+
+  def test_piping_stdin(self):
```

Comment:
How about two different tests here:

`test_compile_stdin()` which compiles and runs hello world from stdin?

and then keep this test but call it `test_preprocess_stdin`?